addon-test-support/ember-qunit/test-isolation-validation.js

```diff
@@ -32,17 +45,10 @@ export function detectIfTestNotIsolated({ module, name }) {
  * @throws Error if tests are not isolated
  */
 export function reportIfTestNotIsolated() {
-  if (TESTS_NOT_ISOLATED.length > 0) {
-    let leakyTests = TESTS_NOT_ISOLATED.slice();
-    TESTS_NOT_ISOLATED.length = 0;
+  if (nonIsolatedTests.hasDebugInfo) {
+    nonIsolatedTests.printToConsole();
+    nonIsolatedTests.reset();
```

Comment:
This is somewhat curious, wouldn't it be easier to do:

```js
nonIsolatedTests = new TestInfoSummary();
```

If we do that, then AFAICT we don't need `reset`?

Reply:
We could that. :)